### ISSUE TITLE
Wire serializer and serialization verification cause StackOverflowException

### ISSUE THREAD

**Issue by akoshelev** [akkadotnet/akka.net#1627]
Hi guys,
I just faced a StackOverflowException in my unit tests. My configuration looks like this
```
akka.actor.serializers.wire = ""Akka.Serialization.WireSerializer, Akka.Serialization.Wire""
akka.actor.serialization-bindings {
    ""System.Object"" = wire
}
akka.actor.serialize-messages = on
```

When I run any of my unit tests I'm getting a ```StackOverflowException``` [here](https://github.com/akkadotnet/akka.net/blob/f3bebf0cc9a99470872001d24cd2b28d46d24fc5/src/core/Akka/Actor/ActorCell.cs#L279)
```csharp
   object deserialized = _systemImpl.Serialization.Deserialize(serialized, serializer.Identifier,
                    message.GetType());
```
It seems that test is completed, several ```DeadLetter``` are send to system actors. During a ```DeadLetter``` deserialization system tries to resolve an actor path (```FromSurrogate``` or ```ToSurrogate``` method does this), but the actor is dead and another ```DeadLetter``` message is created and send and the same process repeats.

**Comment by akoshelev** [comment]
If you can give me an idea how to fix this, I can try to do it myself.

**Comment by rogeralsing** [comment]
Is there a specific message type that fails?
My guess would be that the message is or contains some specific type that is not yet wire friendly.

**Comment by annymsMthd** [comment]
Circular reference?

**Comment by akoshelev** [comment]
here is the code to reproduce it
```csharp
 public class Actor : ReceiveActor
    {
        public Actor()
        {
            Receive<string>(m => Console.WriteLine(m));
        }
    }

    public class Message
    {
        
    }

    class Program
    {
        static void Main(string[] args)
        {
            var conf = @"
                akka.log-dead-letters-during-shutdown = true
                akka.actor.serializers.wire = ""Akka.Serialization.WireSerializer, Akka.Serialization.Wire""
                akka.actor.serialization-bindings {
                  ""System.Object"" = wire
                }
                akka.actor.serialize-messages = on
                akka.actor.serialize-creators = on
                akka.actor.debug.receive = true
                akka.actor.debug.autoreceive = true
                akka.actor.debug.lifecycle = true
                akka.actor.debug.event-stream = true
                akka.actor.debug.unhandled = true
                akka.actor.debug.fsm = true
                akka.actor.debug.router-misconfiguration = true
                akka.actor.guardian-supervisor-strategy = ""Akka.Actor.StoppingSupervisorStrategy""
                akka.loglevel = DEBUG
                akka.log-dead-letters = 10
                akka.stdout-loglevel = DEBUG
            ";
            using (var @as = ActorSystem.Create("actorSystem", ConfigurationFactory.ParseString(conf)))
            {
                var actor = @as.ActorOf(Props.Create(() => new Actor()), "actor");
                actor.Tell("some message");

                @as.Shutdown();
                actor.Tell(new Message());
                @as.AwaitTermination();
                Console.ReadLine();
            }
        }
    }
```

**Comment by Danthar** [comment]
Solved in 1.0.7. As in, I can reproduce this in 1.0.6 but not 1.0.7
Was related to the same problem I discovered in: https://github.com/akkadotnet/akka.net/issues/1724
